internal/store/leaf/fsck.go

```diff
@@ -227,61 +292,61 @@ func (s *Store) fsckCheckEntry(ctx context.Context, name string) (string, ErrorS
 	} else {
 		out.Printf(ctx, "Re-encrypting %s to fix storage format. [leaf store]", name)
 	}
+
 	if err := s.Set(ctxutil.WithGitCommit(ctx, false), name, sec); err != nil {
-		return "", errsFatal, fmt.Errorf("failed to write secret %s: %w", name, err)
+		return "", errs.Append(errsFatal, fmt.Errorf("failed to write secret %s: %w", name, err)).ErrorOrNil()
 	}
 
-	errs, err = s.fsckCheckRecipients(ctx, name)
-	if err != nil {
-		if errs == errsFatal {
-			errcoll += fmt.Errorf("Checking recipients for %s failed:\n    %w", name, err).Error()
+	merr = s.fsckCheckRecipients(ctx, name)
+	if merr.ErrorOrNil() != nil {
+		if merr.Severity == errsFatal {
+			_ = errs.Append(merr.Severity, fmt.Errorf("Checking recipients for %s failed:\n    %w", name, err))
```

Comment:
Do you really want to have `err` here in that `fmt.Errorf`?

Reply:
Thanks. My naming is terrible ...